src/components/WeeklyForecast.svelte

```diff
@@ -0,0 +1,153 @@
+<script>
+  import weatherService from '../services/weatherService';
+  import {
+    Table,
+    TableBody,
+    TableBodyCell,
+    TableBodyRow,
+    TableHead,
+    TableHeadCell,
+  } from 'flowbite-svelte';
+  import moment from 'moment';
+  import { link } from 'svelte-spa-router';
+  import weatherCodes from '../assets/weather-interpretation-code-description.json';
+  import { t } from 'svelte-i18n';
+
+  export let latitude;
+  export let longitude;
+
+  const weatherData = {
+    isLoading: false,
+    hasError: false,
+  };
+  let forecastData = [];
+  const routeBase = '/details';
+
+  function formatDate(dateString) {
+    return moment(dateString)
+      .format('dddd DD MMM');
+  }
+
+  async function fetchWeatherData(latitude, longitude) {
+    weatherData.isLoading = true;
+    weatherData.hasError = false;
+
+      const response = await weatherService.getWeeklyForecast(latitude, longitude);
+      if (!response) {
+        weatherData.isLoading = false;
+        weatherData.hasError = true;
+
+        return;
+      }
+
+      const dailyData = response.daily;
+
+      forecastData = dailyData.time?.map((time, i) => ({
+        date: formatDate(time),
+        rawDate: time,
+        maxTemp: dailyData.temperature_2m_max[i],
+        minTemp: dailyData.temperature_2m_min[i],
+        precipitation: dailyData.precipitation_sum[i],
+        windSpeed: dailyData.wind_speed_10m_max[i],
+        weatherCode: dailyData.weather_code[i],
+      }));
+
+      weatherData.isLoading = false;
+  }
+
+  $: if (latitude && longitude) {
+    fetchWeatherData(latitude, longitude);
+  }
+</script>
+
+<div class="p-5 md:p-10">
+  {#if weatherData.isLoading}
+    <div>{$t('loading')}</div>
+  {:else if weatherData.hasError}
+    <div>{$t('errorLoadingData')}</div>
+  {:else}
+    <div class="block md:hidden">
+      <div class="grid grid-cols-1 gap-4">
+        {#each forecastData as day}
```

Comment:
can you please extract data same way as you did at str 117 
`          {#each forecastData as { date, weatherCode, maxTemp, minTemp, precipitation, windSpeed, rawDate }}
`